app/controllers/courses_controller.rb

```diff
@@ -11,6 +11,13 @@ class CoursesController < ApplicationController
   def index
     load_courses
 
+    @flash_notice_newsletter = :no_newsletter
```

Comment:
After reading the whole PR, I understood what this variable is about. Maybe you could change `:no_newsletter` to something else? If I am correct, this will be set once the user made a decision for or against receiving a newsletter. If unsubscribed_newsletter is false (= he wants to get newsletters), the variable will still be `:no_newsletter`. What about `:no_newsletter_advertising` or `:no_newsletter_flash_notice` etc.?